pkg/asset/cluster/tfvars.go

```diff
@@ -510,17 +510,47 @@ func (t *TerraformVariables) Generate(parents asset.Parents) error {
 			caCert = string(caFile)
 		}
 
-		masters, err := mastersAsset.Machines()
-		if err != nil {
-			return err
+		var masterSpecs []*openstackprovider.OpenstackProviderSpec
+		{
+			masters, err := mastersAsset.Machines()
+			if err != nil {
+				return err
+			}
+
+			for _, master := range masters {
+				masterSpecs = append(masterSpecs, master.Spec.ProviderSpec.Value.Object.(*openstackprovider.OpenstackProviderSpec))
+			}
 		}
 
-		var masterSpecs []*openstackprovider.OpenstackProviderSpec
-		for _, master := range masters {
-			masterSpecs = append(masterSpecs, master.Spec.ProviderSpec.Value.Object.(*openstackprovider.OpenstackProviderSpec))
+		var workerSpecs []*openstackprovider.OpenstackProviderSpec
+		{
+			workers, err := workersAsset.MachineSets()
+			if err != nil {
+				return err
+			}
+
+			for _, worker := range workers {
+				workerSpecs = append(workerSpecs, worker.Spec.Template.Spec.ProviderSpec.Value.Object.(*openstackprovider.OpenstackProviderSpec))
+			}
 		}
+
+		// Only considering the first Compute machinepool here, because
```

Comment:
Let's try to not make assumptions that there will always only be one compute machinepool if possible, because that's going to bite us if the installer lifts the 1 worker machinepool restriction.

Reply:
Dropping this assumption will require us to accept a whole set of new assumptions.

At this stage, we have one MachineSet per Availability zone, and one machine-pool. As a consequence, we have multiple Server group names and one single Server group policy.

If we have multiple machine-pools, there is no straightforward way to match names with policies. The code changes required to support multiple machine-pools are non-negligeable, and I can't justify that effort since there is no proposal to support that case.

I have added an assertion that there is no more than one worker machine-pool. The code will crash loudly at the first test run, and we will clearly see the required change in that case.

WDYT?

Reply:
Fair enough. If it's too complex, it's not worth plan too much ahead and instead adding an obvious error message like you did is perfectly fine.